packages/initiatives/src/util.ts

```diff
@@ -0,0 +1,127 @@
+/* Copyright (c) 2018 Environmental Systems Research Institute, Inc.
+ * Apache-2.0 */
+import { IRequestOptions, getPortalUrl } from "@esri/arcgis-rest-request";
+
+/**
+ *
+ *
+ * @export
+ * @param {string} sourceItemId
+ * @param {string} targetItemId
+ * @param {string} owner
+ * @param {[string]} assets
+ * @param {IRequestOptions} requestOptions
+ * @returns {Promise<boolean>}
+ */
+export function copyImageResources(
+  sourceItemId: string,
+  targetItemId: string,
+  owner: string,
+  assets: string[],
+  requestOptions: IRequestOptions
+): Promise<boolean> {
+  const itemResourceUrl = `${getPortalUrl(
+    requestOptions
+  )}/content/items/${sourceItemId}/resources`;
+  /* istanbul ignore next blob responses are difficult to make cross platform we will just have to trust the isomorphic fetch will do its job */
+  return requestOptions.authentication
+    .getToken(itemResourceUrl)
+    .then(token => {
+      const assetPromises = assets.map(assetName => {
+        const mimeType = getImageMimeTypeFromFileName(assetName);
+        const sourceUrl = `${itemResourceUrl}/${assetName}?token=${token}`;
+        return addImageAsResource(
+          targetItemId,
+          owner,
+          assetName,
+          sourceUrl,
+          mimeType,
+          token,
+          requestOptions
+        );
+      });
+      // This is really more of a fire-and-forget thing, as the Portal API
+      // adds these requests into a queue for processing
+      return Promise.all(assetPromises);
+    })
+    .then(() => {
+      return true;
+    });
+}
+
+/**
+ * Given a filename of an image. return the mime-type
+ *
+ * @export
+ * @param {string} fileName
+ * @returns {string}
+ */
+export function getImageMimeTypeFromFileName(fileName: string): string {
+  let extension = fileName.split(".")[1];
+  // handle jpg
+  if (extension === "jpg") {
+    extension = "jpeg";
+  }
+  return `image/${extension}`;
+}
+
+/**
+ * Load an image from a url, and store it as a resource on an existing item
+ *
```

Comment:
easier/better to use [`@esri/arcgis-rest-items/addItemResource()`](https://esri.github.io/arcgis-rest-js/api/items/addItemResource/)?

Reply:
I tried to refactor this to use the ARGjs methods, but failed utterly.

Reply:
done in 138f6af.